iavl/cli.py

```diff
@@ -376,30 +377,71 @@ def visualize(db, version, store=None, include_prev_version=False):
     type=click.Path(exists=True),
     required=True,
 )
-def dump_changesets(db, start_version, end_version, store: Optional[str], out_dir: str):
+@click.option(
+    "--cache-size",
+    help="the output directory to save the data files",
+    default=DEFAULT_CACHE_SIZE,
+)
+def dump_changesets(
+    db, start_version, end_version, store: Optional[str], out_dir: str, cache_size: int
+):
     """
     extract changeset by comparing iavl versions and save in files
     with compatible format with file streamer.
     end_version is exclusive.
     """
     db = dbm.open(str(db), read_only=True)
     prefix = store_prefix(store) if store is not None else b""
-    ndb = NodeDB(db, prefix=prefix)
-    for _, v, _, changeset in iter_state_changes(
-        db, ndb, start_version=start_version, end_version=end_version, prefix=prefix
-    ):
-        with (Path(out_dir) / f"block-{v}-data").open("wb") as fp:
-            diff.write_change_set(fp, changeset)
+    ndb = NodeDB(db, prefix=prefix, cache_size=cache_size)
+
+    last_version = None
+    offset = 0
+    output = Path(out_dir) / f"block-{start_version}"
```

Comment:
not sure if it's a tradeoff to get rid of small files but we might overwrite output with different `end_version` but with same `start_version`

Reply:
yeah, it's to avoid too many small files, more performant to process and easier to distribute.

> we might overwrite output with different end_version but with same start_version

currently it don't override exiting files, but it could be an issue if the chunk files are not continuously or overlapping, but that's an issue of operation, for example, use a fixed granularity.